email-service/internal/broker/rate/client.go

```diff
@@ -0,0 +1,78 @@
+package rate
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"time"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+)
+
+var rateTimeout = 5 * time.Second
+
+const eventType = "RateFetched"
+
+type rateFetchedEvent struct {
+	broker.Event
+	Data struct {
```

Comment:
maybe extract nested struct definition to a separate type to make it more readable?